plutus-pab/src/Plutus/PAB/Core/ContractInstance.hs

```diff
@@ -227,9 +228,14 @@ stmInstanceLoop def instanceId = do
                 Left () -> do
                     ask >>= liftIO . STM.atomically . InstanceState.setActivity Stopped
                 Right event' -> do
-                    (newState :: Contract.State t) <- Contract.updateContract @t instanceId (caID def) currentState event'
-                    Contract.putState @t def instanceId newState
-                    stmInstanceLoop @t def instanceId
+                    (newState' :: Either PABError (Contract.State t)) <- Contract.updateContract @t instanceId (caID def) currentState event'
+                    case newState' of
+                        Left err -> do
+                            logError @(ContractInstanceMsg t) $ UpdateContractFailed $ show err
```

Comment:
@j-mueller, you have suggested to use `logWarn` here earlier. Is it better in this context?

Because it seems to be an error that happened but we continue to serve the requests. Or ERROR level is only for bugs and errors that terminate the application?

Reply:
Yeah I'd use logError only when the application terminates. The JSON decoding error is something we can easily recover from, without any data loss or anything (apart from the bad request)

Reply:
Aha, your PR #3342 has fixed the bug! I'm closing the issue and this PR then. Thanks! :)

But yeah, it logs with `logError` at the moment:

```
[ERROR] 04876881-31a2-4a55-a2d2-4106b6e2d69e: "OtherError \"Error in $: parsing () failed, expected Array, but encountered Number\""
```